assets/src/dashboard/app/views/editorSettings/publisherLogo/index.js

```diff
@@ -66,39 +71,154 @@ function PublisherLogoSettings({
   publisherLogos,
   uploadError,
 }) {
+  const { isRTL } = useConfig();
+
+  const containerRef = useRef();
+  const gridRef = useRef();
+  const itemRefs = useRef({});
+
+  const [activePublisherLogo, _setActivePublisherLogoId] = useState(null);
+  const [indexRemoved, setIndexRemoved] = useState(null);
+
+  const publisherLogosById = useMemo(() => publisherLogos.map(({ id }) => id), [
+    publisherLogos,
+  ]);
+
+  const publisherLogoCount = useRef(publisherLogosById.length);
+
+  const onRemoveLogoClick = useCallback(
+    (e, { publisherLogo, idx }) => {
+      e.preventDefault();
+
+      handleRemoveLogo(publisherLogo);
+      setIndexRemoved(idx);
+      publisherLogoCount.current = publisherLogosById.length;
+    },
+    [handleRemoveLogo, publisherLogosById.length]
+  );
+
+  const setActivePublisherLogoId = useCallback((id) => {
+    _setActivePublisherLogoId(id);
+  }, []);
```

Comment:
You don't need to wrap the setter with `useCallback` it's reference shouldn't change by just using what is returned from `useState`